### PR TITLE
sequence -> id and make it 32 bytes

### PR DESCRIPTION
**Pull request by kcsongor**
The key motivation behind this change is to make it possible to look up the outbox item on solana from a given id. The incrementally assigned sequence numbers don't work for this, because seeding PDAs by them would create race conditions. By putting the outbox item's account address directly into the payload, this makes it possible to perform those lookups.

### CODE REVIEW

File: evm/src/libraries/TransceiverStructs.sol [hunk 1]
```diff
@@ -26,13 +26,13 @@ library TransceiverStructs {
 
     /// @dev Message emitted and received by the nttManager contract.
     ///      The wire format is as follows:
-    ///      - sequence - 8 bytes
+    ///      - id - 32 bytes
     ///      - sender - 32 bytes
     ///      - payloadLength - 2 bytes
     ///      - payload - `payloadLength` bytes
     struct NttManagerMessage {
         /// @notice unique sequence number
```
Comment (bingyuyap) [inline evm/src/libraries/TransceiverStructs.sol:34]: do we still call this a unique _sequence_ number? since on solana side it's not incremental

Comment (kcsongor) [reply evm/src/libraries/TransceiverStructs.sol:34]: good point, let me fix

File: evm/src/NttManager/NttManager.sol [hunk 2]
```diff
@@ -393,7 +393,9 @@ contract NttManager is INttManager, NttManagerState {
         // construct the NttManagerMessage payload
         bytes memory encodedNttManagerPayload = TransceiverStructs.encodeNttManagerMessage(
             TransceiverStructs.NttManagerMessage(
-                seq, toWormholeFormat(sender), TransceiverStructs.encodeNativeTokenTransfer(ntt)
+                bytes32(uint256(seq)),
```
Comment (djb15) [inline evm/src/NttManager/NttManager.sol:396]: Any particular reason why you decided to cast to uint256 here rather then changing the counter to a uint256? No strong feelings either way personally

**Review by djb15 — APPROVED** [review]

File: evm/src/NttManager/NttManager.sol (hunk 2 above)
Comment (kcsongor) [reply evm/src/NttManager/NttManager.sol:396]: I suppose it's in its own storage slot anyway. would it be crazy to consider packing the sequence counter into the same slot as the outbound rate limit params? they are always accessed together, so would be a nontrivial gas saving.

Comment (djb15) [reply evm/src/NttManager/NttManager.sol:396]: Nice suggestion actually, that could be a follow up PR. Should be a pretty low risk change as gas optimisations go

**Review by RahulMaganti47 — APPROVED** [review]

**Pull request merged**